client/components/Event/EventDetail.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import UpdateEventForm from './UpdateEventForm';
+import { Link } from 'react-router-dom';
+import { getEvents as _getEvents, getEventDetail as _getEventDetail, joinEvent as _joinEvent, unjoinEvent as _unjoinEvent } from '../../actions/events';
+import {
+    Button,
+    Container,
+    Grid,
+    Header,
+    Icon,
+    Image,
+    List,
+    Menu,
+    Segment,
+  } from 'semantic-ui-react';
+
+class EventDetail extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { 
+            going: false,
+            activeItem: 'info',
+            event: {}
+         }
+         this.rsvp = this.rsvp.bind(this);
+         this.unrsvp = this.unrsvp.bind(this);
+         
+    }
+
+    componentDidMount() {
+        this.props.getEventDetail(this.props.match.params.eventId);
+    }
+    
+    componentDidUpdate() {
+        const { event } = this.props;
+        const attendees = event.users;
+        
+
+        if (attendees && this.state.going === false) {
+            const isGoing = attendees.filter(user => user.id === this.props.user.id)
+            if (isGoing.length) {
+                this.setState({going: true})
+            }
+            }
+    }
+
+    rsvp() {
+        this.props.joinEvent(this.props.event)
+        this.setState({going: true})
+    }
+    unrsvp() {
+        this.props.unjoinEvent(this.props.event)
+        this.setState({going: false})
+    }
+
+    
+
+    handleMenuClick = (e, { name }) => this.setState({ activeItem: name })
+
+    render() { 
+        const { event, user, group } = this.props;
+        const { going, activeItem } = this.state;
+        const attendees = event.users;
+        console.log('EVENT', event)
+        console.log('history', this.props.history)
+        return ( 
```

Comment:
We can remove console.logs when we feel confident.

Reply:
went ahead and removed